psydac/linalg/basic.py

```diff
@@ -547,7 +546,7 @@ def __init__(self, domain, codomain, c, A):
         assert isinstance(domain, VectorSpace)
         assert isinstance(codomain, VectorSpace)
         assert np.isscalar(c)
-        assert np.isreal(c)
+        assert(np.isreal(c) or np.iscomplex(c) == (codomain._dtype == complex))
```

Comment:
I am not sure that NumPy's `iscomplex` is the correct function to be used here. According to its docstring, "What is tested is whether the input has a non-zero imaginary part, not if the input type is complex."

Reply:
Fixed in https://github.com/pyccel/psydac/pull/455/commits/db085e6596e413f5ff7df3f8a6d18a59b29e888d